src/App.js

```diff
@@ -1,5 +1,139 @@
+const MissionUtils = require("@woowacourse/mission-utils");
+const inputValidation = require("./validation/inputValidation");
+
 class App {
-  play() {}
+  constructor(inputNum, randomNum, overInputNum) {
+    this.inputNum = inputNum;
+    this.randomNum = randomNum;
+    this.overInputNum = overInputNum;
+  }
+
+  play() {
+    this.userInit();
+    this.gameRoutine();
+  }
+
+  gameRoutine() {
+    this.randomNumber();
+    this.userInput();
+  }
+
+  // userInput logic
+  userInit() {
+    MissionUtils.Console.print("숫자 야구 게임을 시작합니다.");
+  }
+
+  userInput() {
+    MissionUtils.Console.readLine("숫자를 입력해주세요 : ", (input) => {
+      this.inputNum = input;
+      if (!this.checkInputValidation()) {
+        throw new Error("잘못된 값을 입력하셨습니다.");
+      }
+      this.game();
+      if (this.randomNum.join("") == this.inputNum) {
+        this.overMessage();
+        this.overChoice();
+        return;
+      } else this.userInput();
+    });
+  }
+
+  checkInputValidation() {
+    const checkNoOverlap = inputValidation.checkNoOverlap(this.inputNum);
+    const checkOnlyNum = inputValidation.checkOnlyNum(this.inputNum);
+    const checkTreeNum = inputValidation.checkThreeNum(this.inputNum);
+    if (!checkNoOverlap || !checkOnlyNum || !checkTreeNum) {
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
```suggestion
    return (checkNoOverlap && checkOnlyNum && checkTreeNum)
```
느낌표를 if문에 크게 묶으면 조건이 위처럼 될거같은데 그대로 저걸 반환하면 같은 값으로 반환될듯합니다 !